Java/libraries/recognizers-text-datatypes-timex-expression/src/main/java/com/microsoft/recognizers/datatypes/timex/expression/TimexHelpers.java

```diff
@@ -3,320 +3,427 @@
-            switch (timex.getPartOfDay()) {
-                case "DT":
-                    timex = new TimexProperty(TimexCreator.DAYTIME);
-                    break;
-                case "MO":
-                    timex = new TimexProperty(TimexCreator.MORNING);
-                    break;
-                case "AF":
-                    timex = new TimexProperty(TimexCreator.AFTERNOON);
-                    break;
-                case "EV":
-                    timex = new TimexProperty(TimexCreator.EVENING);
-                    break;
-                case "NI":
-                    timex = new TimexProperty(TimexCreator.NIGHT);
-                    break;
-                default:
-                    throw new IllegalArgumentException("unrecognized part of day timerange: timex");
-            }
-        }
-
-        Integer hour = timex.getHour();
-        Integer minute = timex.getMinute();
-        Integer second = timex.getSecond();
-        TimexProperty start = new TimexProperty() {
-            {
-                setHour(hour);
-                setMinute(minute);
-                setSecond(second);
-            }
-        };
-        TimexProperty duration = TimexHelpers.cloneDuration(timex);
-
-        return new TimexRange() {
-            {
-                setStart(start);
-                setEnd(TimexHelpers.timeAdd(start, duration));
-                setDuration(duration);
-            }
-        };
+	public static final HashMap<TimexUnit, String> TimexUnitToStringMap = new HashMap<TimexUnit, String>() {
+		{
+			put(TimexUnit.Year, Constants.TimexYear);
+			put(TimexUnit.Month, Constants.TimexMonth);
+			put(TimexUnit.Week, Constants.TimexWeek);
+			put(TimexUnit.Day, Constants.TimexDay);
+			put(TimexUnit.Hour, Constants.TimexHour);
+			put(TimexUnit.Minute, Constants.TimexMinute);
+			put(TimexUnit.Second, Constants.TimexSecond);
+		}
+	};
+
+	public static final List<TimexUnit> TimeTimexUnitList = Arrays.asList(TimexUnit.Hour, TimexUnit.Minute,
+			TimexUnit.Second);
+
+	public static TimexRange expandDateTimeRange(TimexProperty timex) {
+		HashSet<String> types = timex.getTypes().size() != 0 ? timex.getTypes() : TimexInference.infer(timex);
+
+		if (types.contains(Constants.TimexTypes.DURATION)) {
+			TimexProperty start = TimexHelpers.cloneDateTime(timex);
+			TimexProperty duration = TimexHelpers.cloneDuration(timex);
+			return new TimexRange() {
+				{
+					setStart(start);
+					setEnd(TimexHelpers.timexDateTimeAdd(start, duration));
+					setDuration(duration);
+				}
+			};
+		} else {
+			if (timex.getYear() != null) {
+				TimexRange range = new TimexRange() {
+					{
+						setStart(new TimexProperty() {
+							{
+								setYear(timex.getYear());
+							}
+						});
+						setEnd(new TimexProperty());
+					}
+				};
+				if (timex.getMonth() != null) {
+					range.getStart().setMonth(timex.getMonth());
+					range.getStart().setDayOfMonth(1);
+					range.getEnd().setYear(timex.getYear());
```

Comment:
These `get` are not being assigned anywhere, right? You are retrieving data, but not assigning them

Reply:
If you are trying to update the value, it should be `range.setStart(range.getStart().setDayOfMonth(1));`

Reply:
Why this method was updated? I didn't notice any update in the 2455 about this.

Also @VictorGrycuk , I think that you are right, however, I think that it's confusing as I validated this and works correctly

Reply:
this isn't part of the migration apparently. I'll check what happened here